### PR TITLE
[examples] Add amazon example

### PR DESCRIPTION
**Pull request by jaylees14**
Move amazon example over from `example-templefiles`, now it works 😉

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]

File: examples/amazon.temple [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Product: service {
+  category: ProductCategory;
+  seller: Seller;
+  name: string;
+  description: string;
+  price: float;
+
+  Picture: struct {
+    product: Product;
```
Comment (willjvsmith) [inline examples/amazon.temple]: Can be omitted because of implicit `parentID`

Comment (jaylees14) [reply examples/amazon.temple]: Yes indeed, nice catch

File: examples/amazon.temple [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  customer: Customer;
+  orderDate: datetime;
+
+  // OrderItem is a join table between Orders and Products (many to many)
+  OrderItem: struct {
+    product: Product;
+    #enumerable;
+  }
+
+  #omit([delete]);
```
Comment (c-harding) [inline examples/amazon.temple]: parens are optional with arrays, may look nicer without? Up to you

**Review by c-harding — APPROVED** [review]
lgtm

**Pull request merged**